### PR TITLE
[ch101847] Fix go type aliases

### PR DESCRIPTION
**Pull request by sheenacarswell**
This addresses the issue with parsing Go type aliases.

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -1489,6 +1357,95 @@ func (p *parser) getTypeAsString(fieldType interface{}) string {
 	return fmt.Sprint(fieldType)
 }
 
+func parseStructTags(astField *ast.Field, structSchema *SchemaObject, fieldSchema *SchemaObject, name string) (newName string, skip bool) {
```
Comment (raymonstah) [inline parser.go:1361]: did some refactoring, i didn't write this. just extracted that super long method into a slightly shorter but still long function.

File: parser_test.go [hunk 2]
```diff
@@ -181,7 +118,107 @@ func TestExample(t *testing.T) {
...
+            }
+        }
+    },
+    "components": {
+        "schemas": {
+            "DoubleAlias": {
+                "additionalProperties": {
+                    "type": "object"
+                }
+			},
```
Comment (raymonstah) [inline parser_test.go]: formatting is slightly off, but @kbrackbill does this look right?

File: parser.go [hunk 3]
```diff
@@ -1275,151 +1273,7 @@ astFieldsLoop:
 		} else if isGoTypeOASType(typeAsString) {
 			fieldSchema.Type = goTypesOASTypes[typeAsString]
 		}
-
```
Comment (raymonstah) [inline parser.go:1278]: realizing that this is difficult to review, a lot of it is just refactoring (no behavioral changes)

File: parser.go [hunk 4]
```diff
@@ -1147,8 +1152,16 @@ func (p *parser) parseSchemaObject(pkgPath, pkgName, typeName string) (*SchemaOb
...
+	if isGoTypeOASType(p.getTypeAsString(typeSpec.Type)) && schemaObject.Ref == "" {
+		schemaObject.Type = goTypesOASTypes[p.getTypeAsString(typeSpec.Type)]
+	} else if astIdent, ok := typeSpec.Type.(*ast.Ident); ok {
+		// this is for type aliases to custom types
+		newSchema, err := p.parseSchemaObject(pkgPath, pkgName, astIdent.Name)
+		if err != nil {
+			return nil, err
+		}
+		schemaObject.Properties = newSchema.Properties
+		schemaObject.AdditionalProperties = newSchema.AdditionalProperties
```
Comment (raymonstah) [inline parser.go]: this is the part that I added to parse the double type alias

File: parser_test.go (hunk 2 above)
Comment (kbrackbill) [reply parser_test.go]: looks like this one line is using tabs instead of spaces, you can probably just fix it and it'll be fine

**Review by kbrackbill — APPROVED** [review]
Looks good! Once this is in we can check it in gonfalon and see if that fixes everything or if there are still some more issues.

**Pull request merged**